### PR TITLE
Move sqlparser.Backtick to a new package, "sqlescape"

### PR DESCRIPTION
**Pull request by janneku**
Escaping identifiers is useful in Vitess clients. Rename the function to match the expected use.

This pull request also removes duplicated function in vt worker, in order to prevent SQL injection (it was not escaping backticks in middle of identifiers).

### CODE REVIEW

**Comment by sougou** [comment]
So, I've been thinking that sqltypes has already been getting "fat". So, will it be better to move this function to a new package called `sqlescape`, `sqlids` or `sqlvars`?
I've already been thinking of moving the arithmetic functions out of sqltypes.

The main motivation is that there are people wanting to reuse the RBR code, which needs sqltypes. So, it will be better if it remains lightweight.

I'll wait for @alainjobart and @michael-berlin to chime in.

File: go/sqltypes/query.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+// EscapeID produces a backticked identifier given an input string.
+func EscapeID(in string) string {
+	var buf bytes.Buffer
+	WriteEscapeID(&buf, in)
+	return buf.String()
+}
+
+// WriteEscapeID produces a backticked identifier given an input string.
```
Comment (michael-berlin) [inline go/sqltypes/query.go]: The comment is almost the same.

Maybe s/produces/returns/ for `EscapeID` above and here say that its written to the buffer "buf"?

**Comment by sougou** [comment]
Let's go with sqlescape.

**Comment by janneku** [comment]
SGTM, I'll rewrite this pull request.  Hold on... :)

**Comment by janneku** [comment]
With the power of grep+sed, I bring you a new version. PTAL.

File: go/sqltypes/query.go (hunk 1 above)
Comment (janneku) [reply go/sqltypes/query.go]: Done

**Comment by sougou** [comment]
LGTM

**Pull request merged**